components/style/properties/longhand/position.mako.rs

```diff
@@ -220,3 +220,87 @@ ${helpers.single_keyword("object-fit", "fill contain cover none scale-down",
                               spec="https://drafts.csswg.org/css-grid/#propdef-%s" % longhand,
                               products="gecko")}
 % endfor
+
+<%helpers:longhand name="grid-auto-flow"
+        spec="https://drafts.csswg.org/css-grid/#propdef-grid-auto-flow"
+        products="none"
+        animatable="False">
+    use std::fmt;
+    use style_traits::ToCss;
+    use values::NoViewportPercentage;
+    use values::computed::ComputedValueAsSpecified;
+
+    pub type SpecifiedValue = computed_value::T;
+
+    pub mod computed_value {
+        #[derive(PartialEq, Clone, Eq, Copy, Debug)]
+        #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+        pub enum RowOrColumn {
+            Row,
+            Column,
+        }
+
+        #[derive(PartialEq, Clone, Eq, Copy, Debug)]
+        #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+        pub struct T {
+            pub row_or_column: Option<RowOrColumn>,
```

Comment:
Gecko uses bit fields to check whether `dense` exists along with row/column. Let's achieve this simply with a bool.

``` rust
pub struct T {
    is_row: bool,
    dense: bool,
}
```

This will reduce a lot of unwanted complexity in the code :)